CalPatRec/src/PhiClusterFinder_module.cc

```diff
@@ -0,0 +1,566 @@
+    _minnsh       (config().minNSH()),
+    _threshold    (config().threshold()),
+    _nphiclusters (config().nPhiClusters()),
+    _phimin       (config().phiMin()),
+    _phimax       (config().phiMax()),
+    _nphibins     (config().nPhiBins()),
+    _mindphi      (config().minDPhi()),
+    _pitch        (config().pitch()),
+    _tmin         (config().tMin()),
+    _tmax         (config().tMax()),
+    _tbin         (config().tBin()),
+    _t0calc       (config().t0Calc()),
+    _ymin         (config().yMin()),
+    _maxdt        (config().maxDt()),
+    _minsigma     (config().minSigma()),
+    _npeak        (config().peakWidth()),
+    _chToken      {consumes<ComboHitCollection>(config().ComboHitCollection()) },
+    _tcToken      {consumes<TimeClusterCollection>(config().TimeClusterCollection()) },
+    _useCC        (config().useCC())
+    {
+      art::ServiceHandle<art::TFileService> tfs;
+      _hist1 = tfs->make<TH1F>( "hist1" , "phi spectrum",_nphibins,_phimin,_phimax);
+      unsigned nbinst = (unsigned)rint((_tmax-_tmin)/_tbin);
+      _timespec = TH1F("timespec","time spectrum",nbinst,_tmin,_tmax);
+      produces<TimeClusterCollection>();
+
+      if (_diag != 0) _hmanager = art::make_tool<ModuleHistToolBase>(config().DiagPlugin," ");
+      else            _hmanager = std::make_unique<ModuleHistToolBase>();
+    }
+
+  void PhiClusterFinder::beginJob() {
+    if (_diag > 0) {
+      art::ServiceHandle<art::TFileService> tfs;
+      _hmanager->bookHistograms(tfs);
+    }
+  }
+
+  void PhiClusterFinder::produce(art::Event & event ){
+    _iev = event.id().event();
+    std::unique_ptr<TimeClusterCollection> tccol1(new TimeClusterCollection);
+    auto const& chH = event.getValidHandle(_chToken);
+    _chcol = chH.product();
+
+    // Save the sum phi of a cluster to calculate delta phi between the clusters
+    _phitotal.clear();
+    _nhitotal.clear();
+    _phitotal.resize(_nphiclusters);
+    _nhitotal.resize(_nphiclusters);
+
+    for(int m=0;m<_nphiclusters;m++){
+      _phitotal[m] = 0.;
+      _nhitotal[m] = 0.;
+    }
+    auto const& tcH = event.getValidHandle(_tcToken);
+    const TimeClusterCollection& tccol(*tcH);
+    // For the diagnostics
+    _data._tccol =const_cast<TimeClusterCollection*>(tcH.product());
+
+    if (tccol.size() > 0) {
+      for(size_t ipeak=0; ipeak<tccol.size(); ipeak++) {
+        const auto& tc = tccol[ipeak];
+        int nhits = tc.nhits();
+        _clustno.resize(nhits);
+        for(int k=0; k<nhits; k++) {
+           _clustno[k] = -1;
+        }
+        // Find the time clusters separated in phi
+        findClusters(*tccol1,tc.hits());
+      }
+    }
+    if(_debug > 0 and tccol.size() > 0){
+      int newsize = tccol1->size();
+      int oldsize = tccol.size();
+      int eventno = _iev;
+      printf("Output tccol size = %i Input tccol size =  %i event = %i\n",newsize,oldsize,eventno);
+    }
+
+    TimeClusterCollection& tccolnew(*tccol1);
```

Comment:
if you change the interface of the function `addCaloCluster` you don't need this variable anymore.